src/org/exist/xquery/FunctionCall.java

```diff
@@ -362,15 +362,20 @@ public void accept(ExpressionVisitor visitor) {
         visitor.visitFunctionCall(this);
     }
 
-    private class DeferredFunctionCallImpl extends DeferredFunctionCall {
+    private static class DeferredFunctionCallImpl extends DeferredFunctionCall {
```

Comment:
@adamretter you did not understand the problem and the solution I provide. Please, spend a bit of time to thinking before write proposal. [Point by finger](https://github.com/eXist-db/exist/pull/1271/files/ee056ce36d5885ba5effde23a9bccd1cc5b67854#diff-f2b23bccdd5042ec21691c3e7464d672R443)

That is `private static`, so I do not any issue to access `parent` class variable directly.